### ISSUE TITLE
Validator Strategies

### ISSUE THREAD

**Issue by naterush** [ethereum/cbc-casper#152]
# Issue
Validators need to think for themselves about when to make blocks!

# Proposed Implementation
Clocks w/ different (but consistent speeds), etc. Target a DAG, etc. TODO.

**Comment by SeanAvery** [comment]
A potential problem with validators having different network times is that sends are calculated by `self.time + some delay`. This would seem to get screwy when sent to other validators who have different network times. Maybe the message sends can be based on the receivers network time to get around this.

**Comment by djrtwo** [comment]
The thing about validators having different clocks is that the Network doesn't care. That delay is based on whatever the time is in `Network`, not what the time is in Validator. I send a message, then once some time passes in the network, the other validator can receive it. 

I could imagine more of a push on the receive (network -> validator) with the simulator doing the receives and pushing them to the validators rather than the validator having to constantly ask for new messages.

Am I understanding your comment correctly?

**Comment by SeanAvery** [comment]
Right, right. I get you on network controlling time and not by individual validators. I think **I** misunderstood the original issue statement-- was just pointing out potential problem if validators did have unique time references. 

The current pull model makes sense to me, although I see how the push based approach would be logically simpler. I assume each priority queue would then contain from_validator => to_validator => message. Then all message passes batched from the single network PQ :)
